### PR TITLE
Fixes searching for the source control tool to get the id

### PR DESCRIPTION
**Pull request by clarkerican**
Is part of #186

### CODE REVIEW

File: src/containers/NewProject.js [hunk 1]
```diff
@@ -303,7 +303,7 @@ class NewProject extends Component {
       return
     }
 
-    const sourceControlId = _.findKey(tools.all.allIds, (tool => this.state.githubSelected && tool.name === 'GitHub'))
+    const sourceControlId = _.findKey(tools.all.byId, (tool => this.state.githubSelected && tool.name === 'GitHub'))
```
Comment (jackcmeyer) [inline src/containers/NewProject.js]: What is this underscore business?

**Comment by nischayv** [comment]
@clarkerican Thanks for the fix

File: src/containers/NewProject.js (hunk 1 above)
Comment (nischayv) [reply src/containers/NewProject.js]: It's Lodash

Comment (jackcmeyer) [reply src/containers/NewProject.js]: Ewww

File: src/containers/NewProject.js [hunk 2]
```diff
@@ -316,7 +316,7 @@ class NewProject extends Component {
       license: this.state.dropDownText,
       author: this.state.author,
       version: this.state.version,
-      sourceControl: sourceControlId,
+      sourcecontrol: sourceControlId,
```
Comment (nischayv) [inline src/containers/NewProject.js]: Tyr takes a param called `sourceControl` (According to the README). We should stick with the same names across tyr, endor, and yggdrasil

**Pull request merged**